### PR TITLE
i18n: correct zh-cn translation

### PR DESCRIPTION
**Pull request by KotoriK**
Correct some mistakes in ```lib/zh-cn/diagnosticMessages.generated.json```,  here is the detail of these changes:

* **Incositent with English version.**  ex. current translation of ```A_const_assertions_can_only_be_applied_to_references_to_enum_members_or_string_number_boolean_array__1355``` means ```A 'const' assertion can only be used to refer to enumeration members, strings, numbers, booleans, arrays, or object literals.```
* Correct inappropriate wording, ex. ```literal``` should be translated to ```字面量```, currently it has been translated to ```文本```, ```文字```, which may lead confusing with ```string```.
* Correct grammar error. ex. ```对象字面量可以只指定已知属性``` should be ``` 对象字面量只可以指定已知属性```
* Correct use of capitalization, ex.  ```bigint``` should be ```BigInt```
* Improve consistency of translation. 
* Keywords should not be translated for better understanding. Some of them have been fixed in #46924, but still some are remained. ex. ```require()``` , ```super call``` should be ```super()调用```(currently ```超级调用```)

### CODE REVIEW

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 1]
```diff
@@ -20,11 +20,11 @@
   "A_computed_property_name_in_a_method_overload_must_refer_to_an_expression_whose_type_is_a_literal_ty_1168": "方法重载中的计算属性名称必须引用文本类型或 \"unique symbol\" 类型的表达式。",
   "A_computed_property_name_in_a_type_literal_must_refer_to_an_expression_whose_type_is_a_literal_type__1170": "类型文本中的计算属性名称必须引用类型为文本类型或 \"unique symbol\" 类型的表达式。",
   "A_computed_property_name_in_an_ambient_context_must_refer_to_an_expression_whose_type_is_a_literal_t_1165": "环境上下文中的计算属性名称必须引用类型为文本类型或 \"unique symbol\" 类型的表达式。",
-  "A_computed_property_name_in_an_interface_must_refer_to_an_expression_whose_type_is_a_literal_type_or_1169": "接口中的计算属性名称必须引用必须引用类型为文本类型或 \"unique symbol\" 的表达式。",
+  "A_computed_property_name_in_an_interface_must_refer_to_an_expression_whose_type_is_a_literal_type_or_1169": "接口中的计算属性名称必须引用类型为文本类型或 \"unique symbol\" 的表达式。",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:23]:
```suggestion
  "A_computed_property_name_in_an_interface_must_refer_to_an_expression_whose_type_is_a_literal_type_or_1169": "接口中的计算属性名称所引用的表达式，其类型必须为字面量类型或 \"unique symbol\"。",
```

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 2]
```diff
@@ -20,11 +20,11 @@
...
   "A_computed_property_name_must_be_of_type_string_number_symbol_or_any_2464": "计算属性名的类型必须为 \"string\"、\"number\"、\"symbol\" 或 \"any\"。",
   "A_computed_property_name_of_the_form_0_must_be_of_type_symbol_2471": "窗体“{0}”的计算属性名必须是 \"symbol\" 类型。",
-  "A_const_assertions_can_only_be_applied_to_references_to_enum_members_or_string_number_boolean_array__1355": "\"const\" 断言只能用于引用枚举成员、字符串、数字、布尔值、数组或对象文本。",
-  "A_const_enum_member_can_only_be_accessed_using_a_string_literal_2476": "只有使用字符串文本才能访问常数枚举成员。",
+  "A_const_assertions_can_only_be_applied_to_references_to_enum_members_or_string_number_boolean_array__1355": "\"const\" 断言只能用于引用枚举成员、字符串、数字、布尔值、数组。",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:26]:
```suggestion
  "A_const_assertions_can_only_be_applied_to_references_to_enum_members_or_string_number_boolean_array__1355": "\"const\" 断言只能作用于枚举成员、字符串、数字、布尔值、数组，或对象字面量。",
```

Note that the error code here is not consistent with actual message.

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 3]
```diff
@@ -41,13 +41,13 @@
   "A_dynamic_import_call_returns_a_Promise_Make_sure_you_have_a_declaration_for_Promise_or_include_ES20_2711": "动态导入调用返回 “Promise”。请确保具有对 “Promise” 的声明或在 “--lib” 选项中包含了 “ES2015”。",
   "A_file_cannot_have_a_reference_to_itself_1006": "文件不能引用自身。",
   "A_function_returning_never_cannot_have_a_reachable_end_point_2534": "返回“从不”的函数不能具有可访问的终结点。",
-  "A_function_that_is_called_with_the_new_keyword_cannot_have_a_this_type_that_is_void_2679": "使用 \"new\" 关键字调用的函数不可具有“失效”的 \"this\" 类型。",
+  "A_function_that_is_called_with_the_new_keyword_cannot_have_a_this_type_that_is_void_2679": "使用 \"new\" 关键字调用的函数不可具有\"void\"的 \"this\" 类型。",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:44]:
```suggestion
  "A_function_that_is_called_with_the_new_keyword_cannot_have_a_this_type_that_is_void_2679": "使用 \"new\" 关键字调用的函数的 \"this\" 类型不能为 \"void\"。",
```

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 4]
```diff
@@ -60,14 +60,14 @@
...
-  "A_non_dry_build_would_update_timestamps_for_output_of_project_0_6374": "非干燥生成将更新项目 \"{0}\" 的输出的时间戳",
+  "A_non_dry_build_would_update_output_of_project_0_6375": "非 -dry 生成将更新项目 \"{0}\" 的输出",
+  "A_non_dry_build_would_update_timestamps_for_output_of_project_0_6374": "非 -dry 生成将更新项目 \"{0}\" 的输出的时间戳",
   "A_parameter_initializer_is_only_allowed_in_a_function_or_constructor_implementation_2371": "只允许在函数或构造函数实现中使用参数初始化表达式。",
   "A_parameter_property_cannot_be_declared_using_a_rest_parameter_1317": "不能使用 rest 参数声明参数属性。",
   "A_parameter_property_is_only_allowed_in_a_constructor_implementation_2369": "只允许在构造函数实现中使用参数属性。",
   "A_parameter_property_may_not_be_declared_using_a_binding_pattern_1187": "不能使用绑定模式声明参数属性。",
   "A_path_in_an_extends_option_must_be_relative_or_rooted_but_0_is_not_18001": "“扩展”选项中的路径必须为相对路径或根路径，但“{0}”不是。",
-  "A_promise_must_have_a_then_method_1059": "承诺必须具有 \"then\" 方法。",
+  "A_promise_must_have_a_then_method_1059": "\"Promise\"必须具有 \"then\" 方法。",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:70]:
```suggestion
  "A_promise_must_have_a_then_method_1059": "\"Promise\" 必须具有 \"then\" 方法。",
```

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 5]
```diff
@@ -153,23 +153,23 @@
   "Add_override_modifier_95160": "添加 \"override\" 修饰符",
   "Add_parameter_name_90034": "添加参数名称",
   "Add_qualifier_to_all_unresolved_variables_matching_a_member_name_95037": "将限定符添加到匹配成员名称的所有未解析变量",
-  "Add_this_parameter_95104": "添加“此”参数。",
+  "Add_this_parameter_95104": "添加“this”参数。",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:156]:
```suggestion
  "Add_this_parameter_95104": "添加 \"this\" 参数。",
```

Pointing out for consistency with most other texts

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 6]
```diff
@@ -153,23 +153,23 @@
...
   "Add_undefined_to_a_type_when_accessed_using_an_index_6674": "使用索引访问时，将 “undefined” 添加到类型。",
   "Add_undefined_type_to_all_uninitialized_properties_95029": "将未定义的类型添加到未初始化的所有属性",
-  "Add_undefined_type_to_property_0_95018": "向属性“{0}”添加“未定义”类型",
+  "Add_undefined_type_to_property_0_95018": "向属性“{0}”添加“undefined”类型",
   "Add_unknown_conversion_for_non_overlapping_types_95069": "为非重叠类型添加 \"unknown\" 转换",
   "Add_unknown_to_all_conversions_of_non_overlapping_types_95070": "将 \"unknown\" 添加到非重叠类型的所有转换",
-  "Add_void_to_Promise_resolved_without_a_value_95143": "将 \"void\" 添加到已解析但没有值的承诺",
-  "Add_void_to_all_Promises_resolved_without_a_value_95144": "将 \"void\" 添加到所有已解析但没有值的承诺",
+  "Add_void_to_Promise_resolved_without_a_value_95143": "将 \"void\" 添加到已解析但没有值的Promise",
+  "Add_void_to_all_Promises_resolved_without_a_value_95144": "将 \"void\" 添加到所有已解析但没有值的Promise",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:166]:
```suggestion
  "Add_void_to_all_Promises_resolved_without_a_value_95144": "将 \"void\" 添加到所有没有 resolve 到任何值的 Promise",
```

File: lib/zh-cn/diagnosticMessages.generated.json [hunk 7]
```diff
@@ -293,19 +293,19 @@
...
+  "Cannot_assign_to_private_method_0_Private_methods_are_not_writable_2803": "无法分配给私有方法 \"{0}\"。私有方法不可写。",
   "Cannot_augment_module_0_because_it_resolves_to_a_non_module_entity_2671": "无法扩大模块“{0}”，因为它解析为非模块实体。",
   "Cannot_augment_module_0_with_value_exports_because_it_resolves_to_a_non_module_entity_2649": "无法扩充具有值导出的模块“{0}”，因为它解析为一个非模块的实体。",
   "Cannot_compile_modules_using_option_0_unless_the_module_flag_is_amd_or_system_6131": "无法使用选项“{0}”来编译模块，除非 \"--module\" 标记为 \"amd\" 或 \"system\"。",
   "Cannot_create_an_instance_of_an_abstract_class_2511": "无法创建抽象类的实例。",
   "Cannot_delegate_iteration_to_value_because_the_next_method_of_its_iterator_expects_type_1_but_the_co_2766": "无法将迭代委托到值，因为其迭代器的 \"next\" 方法需要类型 \"{1}\"，但包含它的生成器将始终发送 \"{0}\"。",
   "Cannot_export_0_Only_local_declarations_can_be_exported_from_a_module_2661": "无法导出“{0}”。仅可从模块中导出本地声明。",
   "Cannot_extend_a_class_0_Class_constructor_is_marked_as_private_2675": "无法扩展类“{0}”。类构造函数标记为私有。",
-  "Cannot_extend_an_interface_0_Did_you_mean_implements_2689": "无法扩展接口‘{0}’。您的意思是‘实施’吗？",
+  "Cannot_extend_an_interface_0_Did_you_mean_implements_2689": "无法扩展接口‘{0}’。您是否想使用\"implements\"？",
```
Comment (Josh-Cena) [inline lib/zh-cn/diagnosticMessages.generated.json:308]:
```suggestion
  "Cannot_extend_an_interface_0_Did_you_mean_implements_2689": "无法扩展接口 \"{0}\"。您是否想使用 \"implements\"？",
```

**Review by Josh-Cena — COMMENTED** [review]
Already a big improvement from the original text; I'd like to suggest a few further nits!

**Comment by Josh-Cena** [comment]
I'm also slightly confused by the translation workflow on this repo—I hope this is editing the right file.

**Comment by DanielRosenwasser** [comment]
@csigs @pdostalek can you find a way to upstream these changes?

**Comment by sandersn** [comment]
I finally learned how to refer these changes to Microsoft's internal localisation team. Hopefully they can review them and merge them into the src/loc/lcl/chs files.

For my reference, the two issues are
https://ceapex.visualstudio.com/CEINTL/_workitems/edit/686215
https://ceapex.visualstudio.com/CEINTL/_workitems/edit/685780

**Comment by sandersn** [comment]
The two internal issues were resolved on 18 and 22 August, so I am going to close this PR.

**Pull request closed without merging**